crates/common/src/term.rs

```diff
@@ -98,6 +98,42 @@ pub struct SpinnerReporter {
 }
 
 impl SpinnerReporter {
+    /// Spawns the [`Spinner`] on a new thread with the given message
+    ///
+    /// The spinner's message will be updated via the `reporter` events
+    ///
+    /// On drop the channel will disconnect and the thread will terminate
+    pub fn spawn_with(msg: impl Into<String> + Send + 'static) -> Self {
```

Comment:
why did we need a change in spinner?

Reply:
This is to be able to spawn a spinner with the message `Using Revive [Some version]` during compilation